modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpClient.java

```diff
@@ -44,9 +44,36 @@
    * @return the received {@link HttpResponse}
    * @throws IOException if an error occurs while executing
    * @throws TimeoutException if {@code responseTimeout} is exceeded
+   * @deprecated use {@link #send(HttpRequest, HttpRequestOptions)} instead
    */
-  HttpResponse send(HttpRequest request, int responseTimeout, boolean followRedirects, HttpAuthentication authentication)
-      throws IOException, TimeoutException;
+  @Deprecated
+  default HttpResponse send(HttpRequest request, int responseTimeout, boolean followRedirects, HttpAuthentication authentication)
+      throws IOException, TimeoutException {
+    return send(request, new DefaultHttpRequestOptions(responseTimeout, followRedirects, authentication));
+  }
+
+  /**
+   * Same as {@link #send(HttpRequest, HttpRequestOptions)} but using default options.
+   *
+   * @param request the {@link HttpRequest} to send
+   * @return the received {@link HttpResponse}
+   * @throws IOException if an error occurs while executing
+   * @throws TimeoutException if {@code responseTimeout} is exceeded
+   */
+  default HttpResponse send(HttpRequest request) throws IOException, TimeoutException {
+    return send(request, HttpRequestOptions.builder().build());
+  }
+
+  /**
+   * Sends a HttpRequest blocking the current thread until a response is available or the request times out.
+   *
+   * @param request the {@link HttpRequest} to send
+   * @param options the {@link HttpRequestOptions} to use
+   * @return the received {@link HttpResponse}
+   * @throws IOException if an error occurs while executing
+   * @throws TimeoutException if {@code responseTimeout} is exceeded
+   */
+  HttpResponse send(HttpRequest request, HttpRequestOptions options) throws IOException, TimeoutException;
```

Comment:
add a default implementation that calls the previous `send` method, to keep apis compatible.